clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1373,6 +1373,43 @@ scheduler_tmpl_t<RecordType, ReaderType>::get_input_ordinal(output_ordinal_t out
     return outputs_[output].cur_input;
 }
 
+template <typename RecordType, typename ReaderType>
+int64_t
+scheduler_tmpl_t<RecordType, ReaderType>::get_input_tid(output_ordinal_t output)
+{
+    int index = outputs_[output].cur_input;
+    if (index < 0)
+        return -1;
+    if (inputs_[index].tid == INVALID_THREAD_ID)
+        return inputs_[index].last_record_tid;
+    return inputs_[index].tid;
+}
+
+template <typename RecordType, typename ReaderType>
+int
+scheduler_tmpl_t<RecordType, ReaderType>::get_shard_index(output_ordinal_t output)
+{
+    if (output < 0 || output >= static_cast<output_ordinal_t>(outputs_.size()))
+        return -1;
+    if (TESTANY(sched_type_t::SCHEDULER_USE_INPUT_ORDINALS |
```

Comment:
No, the former only sets the latter flag if there's just one input.

Reply:
```
        /**
         * If there is just one input and just one output stream, this sets
         * #SCHEDULER_USE_INPUT_ORDINALS; otherwise, it has no effect.
         */
```

We document the only effect of SCHEDULER_USE_SINGLE_INPUT_ORDINALS as setting SCHEDULER_USE_INPUT_ORDINALS under a certain condition, and nothing otherwise. If there are more effects, we should change the doc to reflect it.

Reply:
Done, and for SCHEDULER_USE_INPUT_ORDINALS: both point at get_shard_index behavior changes now.